core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -86,9 +87,9 @@ trait RawlsBillingProjectComponent {
       }
     }
 
-    def listProjectsWithServicePerimeter(servicePerimeter: ServicePerimeterName): ReadWriteAction[Seq[RawlsBillingProject]] = {
+    def listProjectsWithServicePerimeterAndStatus(servicePerimeter: ServicePerimeterName, statuses: CreationStatus*): ReadWriteAction[Seq[RawlsBillingProject]] = {
       for {
-        projectRecords <- filter(_.servicePerimeter === servicePerimeter.value).result
+        projectRecords <- filter(rec => rec.servicePerimeter === servicePerimeter.value && rec.creationStatus.inSetBind(statuses.map(_.toString))).result
```

Comment:
What happens if `statuses` is `Nil`?  Seems to me like this request would not return any records.  Just want to double check if that is what we want to happen or not?  My gut says it should either return all records in the specified perimeter or it should throw an error.

Reply:
it will probably error

Reply:
I guess the alternative is to take a collection of statuses so that at the very least, callers would need to pass in an empty collection and thus avoid the NPE.

Reply:
it will not be a NPE, it will be a database exception because an IN expression cannot be empty. The right fix would be to exclude the IN clause if the collection is empty... but right now I would just leave it alone

Reply:
👍